packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts

```diff
@@ -139,6 +156,12 @@ export class EVMAAWallet<B extends EVMBlockchainIncludingTestnet = EVMBlockchain
         }
     }
 
+    //TODO @matias: review this method.
+    // First, I would like to use TransactionRequest from viem instead of ethers.
+    // Second, we need to check if chain supports eip-1559 ro not:
+    // - If it does, we need to send maxFeePerGas and maxPriorityFeePerGas
+    // - If it doesn't, we need to send gasPrice
+    // And with the use of viem TransactionRequest, we can specify the TransactionRequest type (eip1559 or legacy) and be more accurate
```

Comment:
hm yeah, also I'm not sure why this accepts a Deferrable–it makes our life more difficult

Reply:
yes, that's something that takes my attention too